ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/DownstreamPathCalculator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler.sendtransaction;
+
+import java.nio.file.Path;
+
+public class DownstreamPathCalculator {
+
+  private final String httpDownstreamPathPrefix;
+
+  public DownstreamPathCalculator(final String httpDownstreamPathPrefix) {
+    this.httpDownstreamPathPrefix = httpDownstreamPathPrefix;
+  }
+
+  public String calculateDownstreamPath(final String receivedRequestUri) {
+    return Path.of("/", httpDownstreamPathPrefix, receivedRequestUri).toString();
```

Comment:
This probably won't work on Windows as Path.of joins using the FileSystem.getSeperator I think it would be safer to use String.join or something similar

Reply:
updated to be more clever using stringjoiner